WikiWatershed/model-my-watershed
# Add Analyze Front-end for Protected Lands

## Overview

This PR adds the front-end to show results for Protected Lands analysis in the Analyze results component of the application.

The Protected Lands analysis results are added to the "Land" tab. A drop-down was added at the top of the tab contents to switch between viewing analysis results from NLCD and Protected Lands.

Connects #3203

### Demo

![mmw-protected-lands](https://user-images.githubusercontent.com/2320142/72556764-51038f80-386d-11ea-91d5-17108c5c0a0f.gif)

### Notes

There was a refactor that was required to decouple Analysis tasks from the tabs of the Analysis view. There was a tight one-to-one relationship between the model and the UI components. The introduction of the `AnalyzeTaskGroup` model allows for task groups to represent analysis tab results, and for those groups to have one or more tasks/results. This UI refactor allows for the public API to remain unchanged while allowing both NLCD and Protected Lands results be tied to the Lands tab.

## Testing Instructions

 * Checkout this branch and go to the main view.
 * Run an analysis on any area
 * Go to the "Lands" tab. Notice the new dropdown.
 * Switch between NLCD and Protected Lands, notice the contents of the tab change to represent each analysis. The related layer on/off and download CSV work as expected.
 * Test that the HydroShare functionality still works as expected, with a new CSV that represents the protected-lands analysis.
## Checklist

- [ ] _All JavaScript tests pass `./scripts/testem.sh`_ **note** The 'pointsource' analyze test fails for me on both develop and here. Unclear if that's a dev environment issue? Otherwise all tests that pass on develop pass on this branch
